### PR TITLE
Fix to #572 Screen Rotation deletes the user input in the search bar

### PR DESCRIPTION
**Pull request by vansh1sh**
Fixes #[572]
Now search input is retained on rotation
Updated GIF:

### CODE REVIEW

**Comment by srv-twry** [comment]
@vansh1sh Please attach a GIF

**Comment by vansh1sh** [comment]
@srv-twry Okay, I'll make the changes.

**Comment by RohanBh** [comment]
@srv-twry the link you provided above is broken.

**Comment by srv-twry** [comment]
Fixed now.

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 1]
```diff
@@ -66,12 +68,16 @@ protected void onCreate(Bundle savedInstanceState) {
     View contentView = LayoutInflater.from(this).inflate(R.layout.search, null);
     setContentView(contentView);
 
+    if (savedInstanceState != null) {
+        searchText = savedInstanceState.getString("text");
+    } else {
+        searchText = null;
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: No need to set it null. It should be null already.

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.SearchView;
 import android.support.v7.widget.Toolbar;
+import android.util.Log;
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Probably you used Log while debugging the code. Now if there isn't any need for it then please remove this import.

**Comment by vansh1sh** [comment]
@srv-twry Made the changes.

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 3]
```diff
@@ -113,6 +116,11 @@ public boolean onCreateOptionsMenu(Menu menu) {
     MenuItem searchMenuItem = menu.findItem(R.id.menu_search);
     MenuItemCompat.expandActionView(searchMenuItem);
     searchView = (SearchView) menu.findItem(R.id.menu_search).getActionView();
+    if (searchText != null) {
+        searchView.setQuery(searchText,Boolean.FALSE);
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Space after comma.

**Comment by vansh1sh** [comment]
@srv-twry Sorry for that, it's done now.

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 4]
```diff
@@ -66,12 +67,14 @@ protected void onCreate(Bundle savedInstanceState) {
     View contentView = LayoutInflater.from(this).inflate(R.layout.search, null);
     setContentView(contentView);
 
+    if (savedInstanceState != null) {
+        searchText = savedInstanceState.getString("text");
```
Comment (brijeshshah13) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: 2 space indentation

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 5]
```diff
@@ -113,6 +116,11 @@ public boolean onCreateOptionsMenu(Menu menu) {
     MenuItem searchMenuItem = menu.findItem(R.id.menu_search);
     MenuItemCompat.expandActionView(searchMenuItem);
     searchView = (SearchView) menu.findItem(R.id.menu_search).getActionView();
+    if (searchText != null) {
+        searchView.setQuery(searchText, Boolean.FALSE);
```
Comment (brijeshshah13) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: 2 space indentation

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java (hunk 4 above)
Comment (vansh1sh) [reply app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: You meant before "if" statement? I cross checked it, but the indentation seems to be correct, can you please explain it?

Comment (abdulwd) [reply app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Before `searchText` you're using 4 space tab, you should refactor it to 2 space tab.

Comment (vansh1sh) [reply app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Okay @abdulwd @brijeshshah13 got it 👍

**Comment by vansh1sh** [comment]
@brijeshshah13 Sorry, updated it now.

**Comment by brijeshshah13** [comment]
@vansh1sh It would be better if you would squash the commits, 6 commits for changes in a single file isn't good.

**Comment by vansh1sh** [comment]
@brijeshshah13 Done, squashed the commits 👍Check now.

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 6]
```diff
@@ -66,12 +67,14 @@ protected void onCreate(Bundle savedInstanceState) {
     View contentView = LayoutInflater.from(this).inflate(R.layout.search, null);
     setContentView(contentView);
 
+    if (savedInstanceState != null) {
+      searchText = savedInstanceState.getString("text");
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Can we extract "text" to a constant in our Constants file please

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 7]
```diff
@@ -113,6 +116,11 @@ public boolean onCreateOptionsMenu(Menu menu) {
     MenuItem searchMenuItem = menu.findItem(R.id.menu_search);
     MenuItemCompat.expandActionView(searchMenuItem);
     searchView = (SearchView) menu.findItem(R.id.menu_search).getActionView();
+    if (searchText != null) {
+      searchView.setQuery(searchText, Boolean.FALSE);
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Boolean,FALSE? Why not false

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java (hunk 6 above)
Comment (vansh1sh) [reply app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Will make the changes

**Pull request merged**